### PR TITLE
[Glimmer 2] Port `partial` tests

### PR DESCRIPTION
**Pull request by chadhietala**
This supersedes #13199

### CODE REVIEW

**Comment by chadhietala** [comment]
@chancancode @wycats r+?

File: packages/ember-htmlbars/tests/helpers/partial_test.js [hunk 1]
```diff
@@ -1,105 +0,0 @@
...
-  },
-  teardown() {
-    runDestroy(view);
-    runDestroy(owner);
-    Ember.lookup = originalLookup;
-    view = owner = null;
-  }
-});
-
-QUnit.test('should render other templates registered with the container', function() {
```
Comment (chancancode) [inline packages/ember-htmlbars/tests/helpers/partial_test.js]: ✔️ should render other templates registered with the container

File: packages/ember-htmlbars/tests/helpers/partial_test.js [hunk 2]
```diff
@@ -1,105 +0,0 @@
...
-    [OWNER]: owner,
-    template: compile('This {{partial "subTemplateFromContainer"}} is pretty great.')
-  });
-
-  runAppend(view);
-
-  equal(trim(view.$().text()), 'This sub-template is pretty great.');
-});
-
-QUnit.test('should render other slash-separated templates registered with the container', function() {
```
Comment (chancancode) [inline packages/ember-htmlbars/tests/helpers/partial_test.js]: ✔️ should render other slash-separated templates registered with the container

File: packages/ember-htmlbars/tests/helpers/partial_test.js [hunk 3]
```diff
@@ -1,105 +0,0 @@
...
-    [OWNER]: owner,
-    template: compile('This {{partial "child/subTemplateFromContainer"}} is pretty great.')
-  });
-
-  runAppend(view);
-
-  equal(trim(view.$().text()), 'This sub-template is pretty great.');
-});
-
-QUnit.test('should use the current view\'s context', function() {
```
Comment (chancancode) [inline packages/ember-htmlbars/tests/helpers/partial_test.js]: ✔️ should use the current context

File: packages/ember-htmlbars/tests/helpers/partial_test.js [hunk 4]
```diff
@@ -1,105 +0,0 @@
...
-    firstName: 'Kris',
-    lastName: 'Selden'
-  }));
-
-  runAppend(view);
-
-  equal(trim(view.$().text()), 'Who is Kris Selden?');
-});
-
-QUnit.test('Quoteless parameters passed to {{template}} perform a bound property lookup of the partial name', function() {
```
Comment (chancancode) [inline packages/ember-htmlbars/tests/helpers/partial_test.js]: ✔️ Quoteless parameters passed to {{template}} perform a bound property lookup of the partial name

File: packages/ember-glimmer/tests/integration/helpers/partial-test.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    this.runTask(() => set(this.context, 'partialName', 'otherTemplate'));
+
+    this.assertText('This other-template is pretty great.');
+
+    this.runTask(() => set(this.context, 'partialName', null));
+
+    this.assertText('This  is pretty great.');
+
+    this.runTask(() => set(this.context, 'partialName', 'subTemplate'));
```
Comment (chancancode) [inline packages/ember-glimmer/tests/integration/helpers/partial-test.js]: Missing an assertion after this

File: packages/ember-glimmer/tests/integration/helpers/partial-test.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    this.runTask(() => set(this.context, 'firstName', 'Kris'));
+
+    this.assertText('Who is Kris Selden?');
+  }
+
+  ['@htmlbars Quoteless parameters passed to {{partial}} perform a bound property lookup of the partial name']() {
+    this.registerPartial('_subTemplate', 'sub-template');
+    this.registerPartial('_otherTemplate', 'other-template');
+
+    this.render('This {{partial partialName}} is pretty {{partial nonexistent}}great.', { partialName: 'subTemplate' });
```
Comment (chancancode) [inline packages/ember-glimmer/tests/integration/helpers/partial-test.js]: Missing an `assertStableRerender` here

**Comment by chancancode** [comment]
@chadhietala looks great to me other than two missing assertions!

As a follow-up to this (you can send it as a separate PR if you prefer), can you add some tests to confirm whether we support accessing the "local variables" in the current scope inside a partial?

e.g. `{{#with foo as |bar|}}{{partial "triesToUseBarInThePartial}}{{/with}}` or `{{#each foo as |bar|}}{{partial "triesToUseBarInThePartial}}{{/with}}`

**Comment by chadhietala** [comment]
@chancancode Can you kick the sauce labs? Seems to have failed for unrelated reasons?

**Comment by chancancode** [comment]
🏃 kicked

**Comment by chancancode** [comment]
Thank you @chadhietala and @jheth!!

**Pull request merged**